test/e2e/remediation_test.go

```diff
@@ -0,0 +1,477 @@
+	Expect(err).ToNot(HaveOccurred())
+	workerNodeName := workerMachineName
+	vmName := bmhToVmName(workerBmh)
+
+	By("Checking that rebooted node becomes Ready")
+	Logf("Marking a BMH '%s' for reboot", workerBmh.GetName())
+	annotateBmh(ctx, bootstrapClient, workerBmh, rebootAnnotation, pointer.String(""))
+
+	waitForVmsState([]string{vmName}, shutoff, specName)
+
+	waitForNodeStatus(ctx, targetClient, client.ObjectKey{Namespace: "default", Name: workerNodeName}, v1.ConditionUnknown, specName)
+	waitForVmsState([]string{vmName}, running, specName)
+	waitForNodeStatus(ctx, targetClient, client.ObjectKey{Namespace: "default", Name: workerNodeName}, v1.ConditionTrue, specName)
+	monitorNodesStatusGlobalDSL(ctx, targetClient, "default", []string{workerNodeName}, v1.ConditionTrue, specName)
+
+	By("Power cycling worker node")
+	powerCycle(ctx, bootstrapClient, targetClient, bmhToMachineSlice{{
+		baremetalhost: &workerBmh,
+		metal3machine: &workerM3Machine,
+	}}, specName)
+	By("Power cycling 1 control plane node")
+	powerCycle(ctx, bootstrapClient, targetClient, bmhsAndMachines[:1], specName)
+	By("Power cycling 2 control plane nodes")
+	powerCycle(ctx, bootstrapClient, targetClient, bmhsAndMachines[1:3], specName)
+
+	By("Testing unhealthy annotation")
+	newReplicaCount := 1
+	scaleControlPlane(ctx, bootstrapClient, client.ObjectKey{Namespace: "metal3", Name: "test1"}, newReplicaCount)
+
+	Logf("Waiting for 2 BMHs to be Ready")
+	Eventually(func(g Gomega) error {
+		bmhs := getAllBmhs(ctx, bootstrapClient, namespace, specName)
+		g.Expect(filterBmhsByProvisioningState(bmhs, bmh.StateReady)).To(HaveLen(newReplicaCount + len(workerM3Machines)))
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
+
+	Logf("Annotating BMH as unhealthy")
+	annotateBmh(ctx, bootstrapClient, workerBmh, "capi.metal3.io/unhealthy", pointer.String(""))
+
+	By("Deleting a worker machine")
+	workerMachine := getMachine(ctx, bootstrapClient, client.ObjectKey{Namespace: namespace, Name: workerMachineName})
+	Expect(bootstrapClient.Delete(ctx, &workerMachine)).To(Succeed(), "Failed to delete worker Machine")
+
+	Logf("Waiting for worker BMH to be in ready state")
+	Eventually(func(g Gomega) error {
+		g.Expect(bootstrapClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: workerBmh.Name}, &workerBmh)).To(Succeed())
+		g.Expect(workerBmh.Status.Provisioning.State).To(Equal(bmh.StateReady))
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
+
+	Logf("Waiting for 2 BMHs to be Provisioned")
+	Eventually(func(g Gomega) error {
+		bmhs := getAllBmhs(ctx, bootstrapClient, namespace, specName)
+		g.Expect(filterBmhsByProvisioningState(bmhs, bmh.StateProvisioned)).To(HaveLen(2))
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
+
+	By("Scaling up machine deployment")
```

Comment:
let's be specific to make it easier to follow in case of debugging the tests later
```suggestion
	By("Scaling up machine deployment to 3 replicas")
```